### ISSUE TITLE
Is there any better way to re-implement feature extractor?

### ISSUE THREAD

**Issue by Jing-Luo** [linmx0130/ya_mxdet#3]
https://github.com/linmx0130/ya_mxdet/blob/master/faster_rcnn/rpn.py

I guess it is a bit complicated to copy weights from layer to layer. For example, what if we are using resnet 101 as the meta architecture?

Is there any better way to re-implement feature extractor?

**Comment by linmx0130** [comment]
Thanks for your attention!

Well, I think it is possible to make RPNFeatureExtractor as simply a decorator of the base architecture network. The problem is in MXNet VGG pretrained model, the feature part is really "feature". A pooling layer is in the end of the feature part. This terrible design posed a difficult in building general RPNFeatureExactor.

Another problem is in RCNN part. For ResNet, the detection head should be initialized with Gaussian weights, if not the 5th part of the ResNet. 

Do you have any idea?

**Comment by linmx0130** [comment]
I have re-implement the feature extractor! Try it!
But I am not sure it will work correctly on ResNet or other models with batch normalization. Batch normalization layers should use global stats in the training of Faster RCNN since the batch size is 1.
